platform/kube/controller.go

```diff
@@ -399,8 +401,16 @@ func (c *Controller) GetIstioServiceAccounts(hostname string, ports []string) []
 // AppendServiceHandler implements a service catalog operation
 func (c *Controller) AppendServiceHandler(f func(*model.Service, model.Event)) error {
 	c.services.handler.Append(func(obj interface{}, event model.Event) error {
-		if svc := convertService(*obj.(*v1.Service), c.domainSuffix); svc != nil {
-			f(svc, event)
+		svc := *obj.(*v1.Service)
+
+		if svc.Namespace == meta_v1.NamespaceSystem {
```

Comment:
There's a constant defined somewhere that lists the special namespaces such as kube system and kube public. Are we sure we want to exclude those? This might prevent applying routing rules to kubernetes API server, for example.